src/api/telemetry/TelemetryAPI.js

```diff
@@ -368,56 +413,52 @@ export default class TelemetryAPI {
     }
 
     /**
-     * Subscribe to realtime telemetry for a specific domain object.
-     * The callback will be called whenever data is received from a
-     * realtime provider.
+     * Request telemetry staleness for a domain object.
      *
-     * @method subscribe
+     * @method isStale
      * @memberof module:openmct.TelemetryAPI~TelemetryProvider#
      * @param {module:openmct.DomainObject} domainObject the object
-     *        which has associated telemetry
-     * @param {Function} callback the callback to invoke with new data, as
-     *        it becomes available
-     * @returns {Function} a function which may be called to terminate
-     *          the subscription
+     *        which has associated telemetry staleness
+     * @returns {Promise.<object[]>} a promise for a staleness data
+     *  object: ex.
+     *  {
+     *      isStale: <Boolean>,
+     *      timestamp: <timestamp>
+     *  }
      */
-    subscribe(domainObject, callback, options) {
-        const provider = this.#findSubscriptionProvider(domainObject);
+    async isStale(domainObject) {
+        const abortController = new AbortController();
+        const options = { signal: abortController.signal };
 
-        if (!this.subscribeCache) {
-            this.subscribeCache = {};
-        }
+        this.requestAbortControllers.add(abortController);
 
-        const keyString = objectUtils.makeKeyString(domainObject.identifier);
-        let subscriber = this.subscribeCache[keyString];
+        const provider = this.#findStalenessProvider(domainObject);
+        if (!provider) {
+            this.requestAbortControllers.delete(abortController);
```

Comment:
Let's just create the abort controller after this `if` block, rather than creating it and deleting it

Reply:
yeah that makes more sense. done!